src/main/java/com/seatapp/services/SeatService.java

```diff
@@ -112,4 +124,30 @@ public List<Seat> getAllWithReservationsByDate(final LocalDate date) {
         }
         return foundSeats;
     }
+
+    /**
+     * Checks in on the reservation of the seat.
+     *
+     * @param seatId   the seatId from the seat where you check in.
+     * @param username username of the person wanting to check in.
+     */
+    public void checkInOnSeat(final Long seatId, final String username) {
+        Seat seat = getSeatById(seatId);
+
+        Reservation reservation = seat.getReservations().stream()
+                .filter(res -> {
+                    LocalDateTime startCheckInTime = res.getStartTime()
+                            .minusMinutes(Integer
```

Comment:
You can also split `Integer.parseInt(minutesBeforeReservation` up so it becomes more readable.